### PR TITLE
Feature/task 5099 UI tests

### PR DESCRIPTION
**Pull request by ArturFriesen**
## Description
This PR adds UI-Tests for the new contact journal features.

## Jira Ticket
[5099](https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-5099)

### CODE REVIEW

File: src/xcode/ENA/ENA/Source/Scenes/ContactDiary/__tests__/ENAUITests_07_ContactJournalUITests.swift [hunk 1]
```diff
@@ -280,17 +280,148 @@ class ENAUITests_07_ContactJournalUITests: XCTestCase {
...
+		XCTAssertTrue(app.staticTexts["00:42 " + app.localized("ContactDiary_Overview_LocationVisit_Abbreviation_Hours")].exists)
+		XCTAssertTrue(app.staticTexts["Some note!"].exists)
+	}
+
+	func testNavigateToPersonEncounterDayInfo() {
+		app.launchArguments.append(contentsOf: ["-diaryInfoScreenShown", "YES"])
+
+		navigateToJournalOverview()
+
+		// Select 3th cell.
```
Comment (SwissMountainFrog) [inline src/xcode/ENA/ENA/Source/Scenes/ContactDiary/__tests__/ENAUITests_07_ContactJournalUITests.swift:390]:
```suggestion
		// Select 3rd cell.
```

File: src/xcode/ENA/ENA/Source/Scenes/ContactDiary/__tests__/ENAUITests_07_ContactJournalUITests.swift [hunk 2]
```diff
@@ -280,17 +280,148 @@ class ENAUITests_07_ContactJournalUITests: XCTestCase {
...
 		// check count for day entries: 1 add entry cell + 2 locations added
 		XCTAssertEqual(app.descendants(matching: .table).firstMatch.cells.count, 3)
 	}
 
+	func testDetailsSelectionOfPersonEncounter() {
+		app.launchArguments.append(contentsOf: ["-diaryInfoScreenShown", "YES"])
+
+		navigateToJournalOverview()
+
+		// Select 3th cell.
```
Comment (SwissMountainFrog) [inline src/xcode/ENA/ENA/Source/Scenes/ContactDiary/__tests__/ENAUITests_07_ContactJournalUITests.swift:299]:
```suggestion
		// Select 3rd cell.
```

**Review by SwissMountainFrog — CHANGES_REQUESTED** [review]

File: src/xcode/ENA/ENA/Source/Scenes/ContactDiary/__tests__/ENAUITests_07_ContactJournalUITests.swift [hunk 3]
```diff
@@ -280,17 +280,148 @@ class ENAUITests_07_ContactJournalUITests: XCTestCase {
...
+
+		addPersonToDayEntry("Marcus Mustermann")
+
+		// Select details of encounter.
+
+		XCTAssertTrue(app.segmentedControls[AccessibilityIdentifiers.ContactDiaryInformation.Day.maskSituationSegmentedControl].firstMatch.buttons.element(boundBy: 1).waitForExistence(timeout: .medium))
+		app.segmentedControls[AccessibilityIdentifiers.ContactDiaryInformation.Day.maskSituationSegmentedControl].firstMatch.buttons.element(boundBy: 1).tap()
+
+		XCTAssertTrue(app.segmentedControls[AccessibilityIdentifiers.ContactDiaryInformation.Day.settingSegmentedControl].firstMatch.buttons.element(boundBy: 1).waitForExistence(timeout: .medium))
+		app.segmentedControls[AccessibilityIdentifiers.ContactDiaryInformation.Day.settingSegmentedControl].firstMatch.buttons.element(boundBy: 1).tap()
```
Comment (whiskey) [inline src/xcode/ENA/ENA/Source/Scenes/ContactDiary/__tests__/ENAUITests_07_ContactJournalUITests.swift:314]: Improved readbility:
```suggestion
		let maskSituationButton = app.segmentedControls[AccessibilityIdentifiers.ContactDiaryInformation.Day.maskSituationSegmentedControl].firstMatch.buttons.element(boundBy: 1)
		XCTAssertTrue(maskSituationButton.waitForExistence(timeout: .medium))
		maskSituationButton.tap()

		let settingButton = app.segmentedControls[AccessibilityIdentifiers.ContactDiaryInformation.Day.settingSegmentedControl].firstMatch.buttons.element(boundBy: 1)
 		XCTAssertTrue(settingButton.waitForExistence(timeout: .medium))
		settingButton.tap()
```

File: src/xcode/ENA/ENA/Source/Scenes/ContactDiary/__tests__/ENAUITests_07_ContactJournalUITests.swift [hunk 4]
```diff
@@ -280,17 +280,148 @@ class ENAUITests_07_ContactJournalUITests: XCTestCase {
...
+		XCTAssertTrue(app.staticTexts[app.localized("ContactDiary_Day_Encounter_WithoutMask") + ", " + app.localized("ContactDiary_Day_Encounter_Inside")].exists)
+		XCTAssertTrue(app.staticTexts["Some note!"].exists)
+	}
+
+	func testDetailsSelectionOfLocationVisit() {
+		app.launchArguments.append(contentsOf: ["-diaryInfoScreenShown", "YES"])
+
+		navigateToJournalOverview()
+
+		// Select 3th cell.
```
Comment (whiskey) [inline src/xcode/ENA/ENA/Source/Scenes/ContactDiary/__tests__/ENAUITests_07_ContactJournalUITests.swift:337]:
```suggestion
		// Select 3rd cell.
```

File: src/xcode/ENA/ENA/Source/Scenes/ContactDiary/__tests__/ENAUITests_07_ContactJournalUITests.swift [hunk 5]
```diff
@@ -280,17 +280,148 @@ class ENAUITests_07_ContactJournalUITests: XCTestCase {
...
+
+		XCTAssertTrue(app.segmentedControls.firstMatch.waitForExistence(timeout: .medium))
+		app.segmentedControls.firstMatch.buttons[app.localized("ContactDiary_Day_LocationsSegment")].tap()
+
+		// Add location.
+
+		addLocationToDayEntry("Pizzabude")
+
+		// Select details of locaton visit.
+
```
Comment (whiskey) [inline src/xcode/ENA/ENA/Source/Scenes/ContactDiary/__tests__/ENAUITests_07_ContactJournalUITests.swift:352]: Something missing here?

File: src/xcode/ENA/ENA/Source/Scenes/ContactDiary/__tests__/ENAUITests_07_ContactJournalUITests.swift (hunk 4 above)
Comment (whiskey) [reply src/xcode/ENA/ENA/Source/Scenes/ContactDiary/__tests__/ENAUITests_07_ContactJournalUITests.swift:337]: @SwissMountainFrog DEN hast du übersehen! 🥳

**Review by whiskey — CHANGES_REQUESTED** [review]
Kleinkram…

File: src/xcode/ENA/ENA/Source/Scenes/ContactDiary/__tests__/ENAUITests_07_ContactJournalUITests.swift (hunk 5 above)
Comment (ArturFriesen) [reply src/xcode/ENA/ENA/Source/Scenes/ContactDiary/__tests__/ENAUITests_07_ContactJournalUITests.swift]: removed misleading comment.

**Review by SwissMountainFrog — APPROVED** [review]

**Review by whiskey — APPROVED** [review]

**Review by tcfos — APPROVED** [review]

**Pull request merged**